whylog/assistant/__init__.py

```diff
@@ -2,21 +2,39 @@
 
 import six
 
+NOT_IMPLEMENTED_ERROR_MESSAGE = "Subclass should implement this"
+
 
 @six.add_metaclass(ABCMeta)
 class AbstractAssistant(object):
     @abstractmethod
     def add_line(self, line_id, line_object):
-        pass
+        raise NotImplementedError(NOT_IMPLEMENTED_ERROR_MESSAGE)
```

Comment:
this is not how you should do it. Please read https://docs.python.org/2/library/abc.html#abc.abstractmethod and keep the method contents as `pass`. The same goes for the other methods in this class.